terraform/aws/implementation/modules/eks/main.tf

```diff
@@ -1,32 +1,298 @@
+        Effect = "Allow"
+        Principal = {
+          Federated = "arn:aws:iam::${local.account_id}:oidc-provider/${local.oidc_provider}"
+        }
+        Condition = {
+          StringEquals = {
+            "${local.oidc_provider}:aud" = "sts.amazonaws.com"
+            "${local.oidc_provider}:sub" = "system:serviceaccount:${local.namespace}:${local.service_account_name}"
+          }
+        }
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "load_balancer_controller" {
+  role       = aws_iam_role.eks_service_account.name
+  policy_arn = aws_iam_policy.load_balancer_controller.arn
+}
+
+resource "kubectl_manifest" "service_account" {
+  yaml_body = data.kubectl_file_documents.service_account.documents[0]
+}
+
+# kubectl auth to EKS
+resource "terraform_data" "kubeconfig" {
+  depends_on = [module.eks-cluster, kubectl_manifest.cluster_role_binding, kubectl_manifest.service_account]
+
+  provisioner "local-exec" {
+    command = "aws eks update-kubeconfig --name ${module.eks-cluster.cluster_name} --region ${var.region}"
+  }
+}
+
+# cert-manager
+
+# resource "helm_release" "cert_manager" {
+#   depends_on       = [module.eks-cluster]
+#   name             = "cert-manager"
+#   repository       = "https://charts.jetstack.io"
+#   chart            = "cert-manager"
+#   namespace        = "cert-manager"
+#   create_namespace = true
+
+#   set {
+#     name  = "installCRDs"
+#     value = true
+#   }
+
+#   set {
+#     name  = "webhook.securePort"
+#     value = 10260
+#   }
+# }
+
+# Load Balancer Controller
+
+resource "kubectl_manifest" "load_balancer_controller_crds" {
+  depends_on = [module.eks-cluster]
+  for_each   = data.kubectl_file_documents.load_balancer_controller_crds.manifests
+  yaml_body  = each.value
+}
+
+resource "helm_release" "load_balancer_controller" {
+  depends_on = [module.eks-cluster, kubectl_manifest.load_balancer_controller_crds]
+  name       = "aws-load-balancer-controller"
+  repository = "https://aws.github.io/eks-charts"
+  chart      = "aws-load-balancer-controller"
+  namespace  = "kube-system"
+  version    = "1.7.0"
+
+  set {
+    name  = "clusterName"
+    value = module.eks-cluster.cluster_name
+  }
+
+  set {
+    name  = "serviceAccount.create"
+    value = "false"
+  }
+
+  set {
+    name  = "serviceAccount.name"
+    value = local.service_account_name
+  }
+
+  set {
+    name  = "clusterSecretsPermissions.allowAllSecrets"
+    value = "true"
+  }
+
+  set {
+    name  = "region"
+    value = var.region
+  }
+
+  set {
+    name  = "vpcId"
+    value = var.vpc_id
+  }
+
+  set {
+    name  = "enableCertManager"
+    value = "false"
+  }
+}
+
+
+# Sleep for 120 seconds to allow the load balancer controller to start
+resource "terraform_data" "wait_for_load_balancer_controller" {
+  depends_on = [helm_release.load_balancer_controller]
+
+  provisioner "local-exec" {
+    command = "sleep 120"
+  }
+}
+
+# Building blocks
+
+resource "helm_release" "building_blocks" {
```

Comment:
This `helm_release` resource has a lot of duplicate code between the AWS and Azure instance 🤔 Any way to share it between the two?

Reply:
Hmmm, I could potentially move the code to a separate module that we could import on both sides, yeah. Would require some reorganization though. Might like to keep it as a tech debt ticket for later